### PR TITLE
feat(utils): add an utility function to find string prefix by separator

### PR DESCRIPTION
**Pull request by WHBANG**
### What problem does this PR solve? 
https://github.com/apache/incubator-pegasus/issues/1337

- Unit test

### CODE REVIEW

**Review by acelyc111 — DISMISSED** [review]

File: src/utils/strings.cpp [hunk 1]
```diff
@@ -417,5 +417,14 @@ std::string string_md5(const char *buffer, unsigned length)
 
     return result;
 }
+
+std::string find_string_prefix(const std::string &input, char separator)
+{
+    std::size_t current = input.find(separator);
```
Comment (empiredan) [inline src/utils/strings.cpp:423]:
```suggestion
    auto current = input.find(separator);
```

File: src/utils/strings.cpp [hunk 2]
```diff
@@ -417,5 +417,14 @@ std::string string_md5(const char *buffer, unsigned length)
...
+    if (current == 0 || current == std::string::npos) {
+        return "";
```
Comment (empiredan) [inline src/utils/strings.cpp:425]: Initializing with c-style empty string literal may be less efficient for compilers with older version even if optimization is enabled. Use default constructor instead:
```suggestion
        return std::string();
```

File: src/utils/test/utils.cpp [hunk 3]
```diff
@@ -443,6 +443,26 @@ TEST(core, dlink)
...
 }
 
+TEST(core, find_string_prefix)
+{
+    struct test_case
+    {
+        std::string input;
+        char separator;
+        std::string expected_prefix;
+    } tests[] = {{"", ' ', ""},
```
Comment (empiredan) [inline src/utils/test/utils.cpp:453]: Could add cases that the separator is just the whole source string, and the source string only include separators, such as:
```c++
{{" ", ' ', ""}},
{{"..", '.', ""}},
```

File: src/utils/test/utils.cpp [hunk 4]
```diff
@@ -443,6 +443,26 @@ TEST(core, dlink)
...
+                 {"abc.def", ' ', ""},
+                 {"abc.def", '.', "abc"},
+                 {"ab.cd.ef", '.', "ab"},
+                 {"abc...def", '.', "abc"},
+                 {".abc.def", '.', ""}};
+    std::string actual_output;
+    for (const auto &test : tests) {
+        actual_output = find_string_prefix(test.input, test.separator);
```
Comment (empiredan) [inline src/utils/test/utils.cpp:461]:
```suggestion
        auto actual_output = find_string_prefix(test.input, test.separator);
```

**Review by empiredan — APPROVED** [review]

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**